x/bridge/keeper/params.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/osmoutils"
+	"github.com/osmosis-labs/osmosis/v23/x/bridge/types"
+)
+
+type UpdateParamsResult struct {
+	signersToCreate []string
+	signersToDelete []string
+	assetsToCreate  []types.AssetWithStatus
+	assetsToDelete  []types.AssetWithStatus
+}
+
+// UpdateParams properly updates params of the module.
+func (k Keeper) UpdateParams(ctx sdk.Context, newParams types.UpdateParams) UpdateParamsResult {
+	var (
+		newAssets = osmoutils.Map(newParams.Assets, func(v types.Asset) types.AssetWithStatus {
+			return types.AssetWithStatus{
+				Asset:       v,
+				AssetStatus: types.AssetStatus_ASSET_STATUS_BLOCKED_BOTH,
+			}
+		})
+
+		oldParams = k.GetParams(ctx)
+
+		signersToCreate = osmoutils.Difference(newParams.Signers, oldParams.Signers)
+		signersToDelete = osmoutils.Difference(oldParams.Signers, newParams.Signers)
+		assetsToCreate  = osmoutils.Difference(newAssets, oldParams.Assets)
+		assetsToDelete  = osmoutils.Difference(oldParams.Assets, newAssets)
+	)
```

Comment:
Q: why using a `var` block and not `x := y`. I believe that would improve readability

Reply:
I want to use all features of the language I use :)

But seriously, I just like more this way. I deleted osmoutils.Map usage on line 21, so I hope now this declaration looks clean and structured.